### ISSUE TITLE
Incorrect imports when targeting es2015 using ngc in Ng 4.x

### ISSUE THREAD

**Issue by alvaro450** [angular/angular#17171]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

The particular issue I'm having is with ramda, but I believe it could apply to any other library that does not provide typings where we have to add the typings from @types.

**Current behavior**

When setting the target on tsconfig.app.json to es2015, the output code's import statement for ramda library is incorrect:

import {uniq} from 'ramda/index';

**Expected behavior**

The import statement in the output file should be:

import {uniq} from 'ramda';

**Minimal reproduction of the problem with instructions**

I created a new app using the cli, and just changing the target on the tsconfig.app.json file to es2015. When you run npm run release, the build/app/app.component.js file contains the incorrect path to ramda.

Here is the repo to use to easily reproduce the issue:
https://github.com/alvaro450/typings-issue

**What is the motivation / use case for changing the behavior?**

When creating a library, having the incorrect import statement would mean that when I use the library in an app, it will fail when it tries to resolve that import.

**Please tell us about your environment:**

MacOs Sierra, Visual Studio Code, npm 5

* **Angular version:** ^4.0.0

It is still an issue.

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 Browser independent, since this is the build process.

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript

* **Node (for AoT issues):** `node --version` =   
6.10.0

**Comment by battmanz** [comment]
I'm having the same issue with lodash-es. It's transpiling this:
```ts
import { trimEnd, trimStart } from 'lodash-es';
```
to this
```js
import { trimEnd, trimStart } from 'lodash-es/index';
```
Note that if I transpile my code with `tsc` instead of `ngc`, then I get the correct output. So the problem is with `ngc`, not `tsc`.

**Comment by merrillpaul** [comment]
Same with hammerjs, iscroll, Ngx-charts

**Comment by charlouze** [comment]
I'm having the same issue...

**Comment by merrillpaul** [comment]
To get around this issue, I  forced `@angular/compiler-cli` to `4.1.3` while I let all the other angular artifacts to get `4.2.2`. So far things are working, albeit I have not tested aot builds. For me, I was creating a reusable angular module library for our private npm registry. Building that lib with anything above 4.1.3 compiler-cli was creating these incorrect imports with `/index` suffixed.

**Comment by gms1** [comment]
using @angular/compiler-cli@4.1.3 and 4.2.6 for others didn't work for me; the same '/index' suffix has been added

**Comment by jhwegener** [comment]
@merrillpaul thx! that helps.

**Comment by merrillpaul** [comment]
Another way to use the latest compiler was to do some post processing like https://github.com/merrillpaul/angular-md-pattern-lib/blob/master/lib/gulpfile.js#L152

**Comment by mattem** [comment]
@chuckjaz Is there any update on this issue? It's currently stopping our team from updating angular and angular-cli. Using the latest CLI with `@angular/compiler-cli` pinned to `4.1.3` causes `ERROR in this.compiler.compileAll is not a function`.

**Comment by jackjoy** [comment]
I encountered this problem serval months ago when i build a angular library,finally i use a workaround to resolve this issue.Change the entry file name from 'index' to other name,e.g, change index.ts to main.ts , or change index.d.ts to main.d.ts will do the trick.

**Comment by gms1** [comment]
The workaround proposed by merrillpaul is easy to implement and is especially helpful when using third party libraries ;-)

**Comment by aminpaks** [comment]
I wrote [this rollup plugin](https://www.npmjs.com/package/rollup-plugin-absolute-module-fix) that will fix it temporarily, looking forward to see you guys fix it :)

**Comment by tbosch** [comment]
Note: In ng 5, this is fixed via https://github.com/angular/angular/commit/f24ea59f74552113f8e8586d2d69045254512aa1

**Comment by gtranter62** [comment]
I'm facing the same issue with amcharts3-angular. The problem appears to affect es5 as well as es2015.

**Comment by gtranter62** [comment]
@aminpaks Thanks for the plugin - works great!

**Comment by marcelomelodf** [comment]
same here, importing from codemirror results in `import { ... } from 'codemirror/index'`, except there's no index.js in codemirror's package

**Comment by tbosch** [comment]
This landed on 4.4 as c9f8718d2a72640da3889ac0ddf47a40ee627f67
